frontend/src/data/query/MetricQuery.ts

```diff
@@ -65,10 +65,21 @@ export class MetricQueryResponse {
     return this.invalidValues[fieldName] === this.data.length;
   }
 
-  getUniqueFieldValues(fieldName: string): string[] {
-    if (this.isFieldMissing(fieldName)) {
-      return [];
+  // Calculate numerical range for a field or return undefined if not applicable
+  getFieldRange(fieldName: string): FieldRange | undefined {
+    const fieldValues = this.data
+      .filter((row) => !isNaN(row[fieldName]))
+      .map((row) => row[fieldName]);
+    if (fieldValues.length === 0) {
+      return undefined;
     }
+    return {
+      min: Math.min.apply(Math, fieldValues),
+      max: Math.max.apply(Math, fieldValues),
+    };
+  }
+
+  getUniqueFieldValues(fieldName: string): string[] {
```

Comment:
Did you mean to drop the `if (this.isFieldMissing(fieldName)) {` check?

Reply:
oops good catch